app/controllers/results_controller.rb

```diff
@@ -50,11 +50,14 @@ def redirect_to_c_sharp
 
     query_string = request.query_string
     base_url = Settings.search_and_compare_ui.base_url
-
     redirect_uri = URI(base_url)
     redirect_uri.path = "/results"
     redirect_uri.query = query_string.gsub("%2C", ",") if query_string.present?
 
     redirect_to redirect_uri.to_s
   end
+
+  def subjects_params_array
+    params["subjects"].split(",").map(&:to_s)
```

Comment:
```suggestion
    params["subjects"].split(",")
```
`.split()` only ever returns an array of strings